test/common/test-error-reporter.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+const { relative } = require('node:path');
+const { inspect } = require('node:util');
+const cwd = process.cwd();
+
+module.exports = async function* errorReporter(source) {
+  for await (const event of source) {
+    if (event.type === 'test:fail') {
+      const { name, details, line, column, file } = event.data;
+      let { error } = details;
+
+      if (error?.failureType === 'subtestsFailed') {
+        // In the interest of keeping things concise, skip failures that are
+        // only due to nested failures.
+        continue;
+      }
+
+      if (error?.code === 'ERR_TEST_FAILURE') {
+        error = error.cause;
+      }
+
+      const output = [
+        `Test failure: '${name}'`,
+      ];
+
+      if (file) {
+        output.push(`Location: ${relative(cwd, file)}:${line}:${column}`);
+      }
+
+      output.push(inspect(error));
+      output.push('\n');
+      yield output.join('\n');
+
+      if (process.env.FAIL_FAST) {
+        yield `\n\u001b[31m✖ Bailing on failed test: ${event.data.name}\u001b[0m\n`;
```

Comment:
I think we should avoid any special characters here just because some parts of Node's Jenkins setup don't them properly.

```suggestion
        yield `\nBailing on failed test: ${event.data.name}\n`;
```